### PR TITLE
Add property to config form for zookeeperConfigDirectory

### PR DESCRIPTION
**Pull request by BenWhitehead**

### CODE REVIEW

**Comment by alberts** [comment]
lgtm. maybe send this upstream to https://github.com/soabase/exhibitor ?

File: exhibitor-core/src/main/resources/com/netflix/exhibitor/core/ui/index.html [hunk 1]
```diff
@@ -165,6 +165,7 @@
                         <legend class="caution-legend">Paths</legend>
                         <label for="config-zookeeper-install-dir">ZooKeeper Install Dir</label><input type="text" id="config-zookeeper-install-dir" name="config-zookeeper-install-dir" size="60" title="The directory where the ZooKeeper server is installed">
                             <button id="config-zookeeper-install-dir-help-button">Read Me</button><br clear="all"/>
+                        <label for="config-zookeeper-config-dir">ZooKeeper Configuration Dir</label><input type="text" id="config-zookeeper-config-dir" name="config-zookeeper-config-dir" size="60" title="The directory where the ZooKeeper config is stored. Leave blank to use the ZooKeeper Snapshot Dir"><br clear="all"/>
```
Comment (lingmann) [inline exhibitor-core/src/main/resources/com/netflix/exhibitor/core/ui/index.html]: Do we want a `<span class="optional">optional</span>` here similar to the way `config-zookeeper-log-dir`is done below?

**Comment by lingmann** [comment]
@orsenthil what do you think about getting this project wired up to merge-bot?

File: exhibitor-core/src/main/resources/com/netflix/exhibitor/core/ui/index.html (hunk 1 above)
Comment (BenWhitehead) [inline exhibitor-core/src/main/resources/com/netflix/exhibitor/core/ui/index.html]: I don't think so, the config directory has to have a value if it doesn't then we end up in the same scenario as the original bug.

**Comment by BenWhitehead** [comment]
## Manually tested

### Environment
* DC/OS 1.7.3 -> DC/OS master (including this patch)
* Bootstrap zk as exhibitor backend
* Aws VPC, ec2 instances running CentOS 7.2

#### config.yaml
```
bootstrap_url: http://10.10.0.233
cluster_name: 'ben-test'
customer_key: 35ab8b19-21b3-49d5-be6d-258edb818028
exhibitor_storage_backend: zookeeper
exhibitor_zk_hosts: 10.10.0.233:2181
exhibitor_zk_path: /dcos-exhibitor-bootstrap
master_discovery: static
master_list:
- 10.10.0.232
- 10.10.0.231
- 10.10.0.230
resolvers:
- 8.8.4.4
- 8.8.8.8
# Choose your security mode: permissive, strict, or disabled
#security: disabled
superuser_password_hash: $6$rounds=656000$y6jlyWKK7WNcemV/$gmJMctUNEVFpaUj3TG9VLoPdIspcGmfx0vI2s4fuex4jNE1dBane7r8x9MZ9r7ge8VzXFuMHFFEkUfkxEazI40
superuser_username: bootstrapuser
```


### Testing instructions
1. Spin up VPC with 4 machines running CentOS 7.2
2. Following instructions in [advanced installer](https://docs.mesosphere.com/1.7/administration/installing/custom/advanced/) install DC/OS 1.7.3
3. Wait for all masters to boot, exhibitor to start all zk nodes and leader to be elected
4. From bootstrap zk save z-node `/dcos-exhibitor-bootstrap/configs/config`
5. Following instructions in [upgrading](https://docs.mesosphere.com/1.8/administration/upgrading/#disable), upgrade DC/OS to new build that includes this patch
6. Wait for all masters to come up and turn green in the exhibitor UI
7. From bootstrap zk save z-node `/dcos-exhibitor-bootstrap/configs/config`
8. Update exhibitor config
    1. Force refresh the exhibitor UI to ensure new UI is loaded
    2. In exhibitor UI click on config tab at the top
    3. Toggle the editing switch to 'On'
    4. Update the following field to have the new correct values

        |   |   |
        |---|---|
        | ZooKeeper Snapshot Dir    | `/var/lib/dcos/exhibitor/zookeeper/snapshot` |
        | ZooKeeper Transaction Dir | `/var/lib/dcos/exhibitor/zookeeper/transactions` |
        | Log Index Dir             | `/var/lib/dcos/exhibitor/zookeeper/transactions` |
    5. Scroll back up to the top of the page and click `Commit...`
    6. Click `Rolling Release...`
    7. Click `OK`
    8. Click `OK` to close the alert dialog
9. Wait for the `rolling configuration release` modal to complete
10. From bootstrap zk save z-node `/dcos-exhibitor-bootstrap/configs/config`
11. Notice new values are maintained



### Appendix

Important property of note in the below dumps is `com.netflix.exhibitor-rolling.zookeeper-config-directory`

#### ex_config_1.7.3.properties
```
#Auto-generated by Exhibitor 10.10.0.232
#Tue Nov 22 21:43:56 UTC 2016
com.netflix.exhibitor-rolling-hostnames=
com.netflix.exhibitor-rolling.zookeeper-data-directory=/var/lib/zookeeper/snapshot
com.netflix.exhibitor-rolling.servers-spec=1\:10.10.0.230,2\:10.10.0.231,3\:10.10.0.232
com.netflix.exhibitor.java-environment=
com.netflix.exhibitor.zookeeper-data-directory=/var/lib/zookeeper/snapshot
com.netflix.exhibitor-rolling-hostnames-index=0
com.netflix.exhibitor-rolling.java-environment=
com.netflix.exhibitor-rolling.observer-threshold=0
com.netflix.exhibitor.servers-spec=1\:10.10.0.230,2\:10.10.0.231,3\:10.10.0.232
com.netflix.exhibitor.cleanup-period-ms=300000
com.netflix.exhibitor.auto-manage-instances-fixed-ensemble-size=3
com.netflix.exhibitor.zookeeper-install-directory=/opt/mesosphere/active/exhibitor/usr/zookeeper
com.netflix.exhibitor.check-ms=30000
com.netflix.exhibitor.zookeeper-log-directory=/var/lib/zookeeper/transactions
com.netflix.exhibitor-rolling.auto-manage-instances=1
com.netflix.exhibitor-rolling.cleanup-period-ms=300000
com.netflix.exhibitor-rolling.auto-manage-instances-settling-period-ms=0
com.netflix.exhibitor-rolling.check-ms=30000
com.netflix.exhibitor.log-index-directory=/var/lib/zookeeper/transactions
com.netflix.exhibitor-rolling.log-index-directory=/var/lib/zookeeper/transactions
com.netflix.exhibitor.backup-period-ms=600000
com.netflix.exhibitor-rolling.connect-port=2888
com.netflix.exhibitor-rolling.election-port=3888
com.netflix.exhibitor-rolling.backup-extra=
com.netflix.exhibitor.client-port=2181
com.netflix.exhibitor-rolling.zoo-cfg-extra=tickTime\=2000&initLimit\=10&syncLimit\=5&quorumListenOnAllIPs\=true&maxClientCnxns\=0&autopurge.snapRetainCount\=5&autopurge.purgeInterval\=6
com.netflix.exhibitor-rolling.zookeeper-install-directory=/opt/mesosphere/active/exhibitor/usr/zookeeper
com.netflix.exhibitor.cleanup-max-files=20
com.netflix.exhibitor-rolling.auto-manage-instances-fixed-ensemble-size=3
com.netflix.exhibitor-rolling.backup-period-ms=600000
com.netflix.exhibitor-rolling.client-port=2181
com.netflix.exhibitor.backup-max-store-ms=21600000
com.netflix.exhibitor-rolling.cleanup-max-files=20
com.netflix.exhibitor-rolling.backup-max-store-ms=21600000
com.netflix.exhibitor.connect-port=2888
com.netflix.exhibitor.backup-extra=
com.netflix.exhibitor.observer-threshold=0
com.netflix.exhibitor.log4j-properties=
com.netflix.exhibitor.auto-manage-instances-apply-all-at-once=1
com.netflix.exhibitor.election-port=3888
com.netflix.exhibitor-rolling.auto-manage-instances-apply-all-at-once=1
com.netflix.exhibitor.zoo-cfg-extra=tickTime\=2000&initLimit\=10&syncLimit\=5&quorumListenOnAllIPs\=true&maxClientCnxns\=0&autopurge.snapRetainCount\=5&autopurge.purgeInterval\=6
com.netflix.exhibitor-rolling.zookeeper-log-directory=/var/lib/zookeeper/transactions
com.netflix.exhibitor.auto-manage-instances-settling-period-ms=0
com.netflix.exhibitor-rolling.log4j-properties=
com.netflix.exhibitor.auto-manage-instances=1
```

#### ex_config_NEW-pre.properties
```
#Auto-generated by Exhibitor 10.10.0.232
#Tue Nov 22 21:43:56 UTC 2016
com.netflix.exhibitor-rolling-hostnames=
com.netflix.exhibitor-rolling.zookeeper-data-directory=/var/lib/zookeeper/snapshot
com.netflix.exhibitor-rolling.servers-spec=1\:10.10.0.230,2\:10.10.0.231,3\:10.10.0.232
com.netflix.exhibitor.java-environment=
com.netflix.exhibitor.zookeeper-data-directory=/var/lib/zookeeper/snapshot
com.netflix.exhibitor-rolling-hostnames-index=0
com.netflix.exhibitor-rolling.java-environment=
com.netflix.exhibitor-rolling.observer-threshold=0
com.netflix.exhibitor.servers-spec=1\:10.10.0.230,2\:10.10.0.231,3\:10.10.0.232
com.netflix.exhibitor.cleanup-period-ms=300000
com.netflix.exhibitor.auto-manage-instances-fixed-ensemble-size=3
com.netflix.exhibitor.zookeeper-install-directory=/opt/mesosphere/active/exhibitor/usr/zookeeper
com.netflix.exhibitor.check-ms=30000
com.netflix.exhibitor.zookeeper-log-directory=/var/lib/zookeeper/transactions
com.netflix.exhibitor-rolling.auto-manage-instances=1
com.netflix.exhibitor-rolling.cleanup-period-ms=300000
com.netflix.exhibitor-rolling.auto-manage-instances-settling-period-ms=0
com.netflix.exhibitor-rolling.check-ms=30000
com.netflix.exhibitor.log-index-directory=/var/lib/zookeeper/transactions
com.netflix.exhibitor-rolling.log-index-directory=/var/lib/zookeeper/transactions
com.netflix.exhibitor.backup-period-ms=600000
com.netflix.exhibitor-rolling.connect-port=2888
com.netflix.exhibitor-rolling.election-port=3888
com.netflix.exhibitor-rolling.backup-extra=
com.netflix.exhibitor.client-port=2181
com.netflix.exhibitor-rolling.zoo-cfg-extra=tickTime\=2000&initLimit\=10&syncLimit\=5&quorumListenOnAllIPs\=true&maxClientCnxns\=0&autopurge.snapRetainCount\=5&autopurge.purgeInterval\=6
com.netflix.exhibitor-rolling.zookeeper-install-directory=/opt/mesosphere/active/exhibitor/usr/zookeeper
com.netflix.exhibitor.cleanup-max-files=20
com.netflix.exhibitor-rolling.auto-manage-instances-fixed-ensemble-size=3
com.netflix.exhibitor-rolling.backup-period-ms=600000
com.netflix.exhibitor-rolling.client-port=2181
com.netflix.exhibitor.backup-max-store-ms=21600000
com.netflix.exhibitor-rolling.cleanup-max-files=20
com.netflix.exhibitor-rolling.backup-max-store-ms=21600000
com.netflix.exhibitor.connect-port=2888
com.netflix.exhibitor.backup-extra=
com.netflix.exhibitor.observer-threshold=0
com.netflix.exhibitor.log4j-properties=
com.netflix.exhibitor.auto-manage-instances-apply-all-at-once=1
com.netflix.exhibitor.election-port=3888
com.netflix.exhibitor-rolling.auto-manage-instances-apply-all-at-once=1
com.netflix.exhibitor.zoo-cfg-extra=tickTime\=2000&initLimit\=10&syncLimit\=5&quorumListenOnAllIPs\=true&maxClientCnxns\=0&autopurge.snapRetainCount\=5&autopurge.purgeInterval\=6
com.netflix.exhibitor-rolling.zookeeper-log-directory=/var/lib/zookeeper/transactions
com.netflix.exhibitor.auto-manage-instances-settling-period-ms=0
com.netflix.exhibitor-rolling.log4j-properties=
com.netflix.exhibitor.auto-manage-instances=1
```

#### ex_config_NEW-post.properties
```
#Auto-generated by Exhibitor 10.10.0.232
#Tue Nov 22 22:13:31 UTC 2016
com.netflix.exhibitor-rolling-hostnames=
com.netflix.exhibitor-rolling.zookeeper-data-directory=/var/lib/dcos/exhibitor/zookeeper/snapshot
com.netflix.exhibitor-rolling.servers-spec=1\:10.10.0.230,2\:10.10.0.231,3\:10.10.0.232
com.netflix.exhibitor.zookeeper-pid-path=
com.netflix.exhibitor.java-environment=
com.netflix.exhibitor.zookeeper-data-directory=/var/lib/dcos/exhibitor/zookeeper/snapshot
com.netflix.exhibitor-rolling-hostnames-index=0
com.netflix.exhibitor-rolling.java-environment=
com.netflix.exhibitor-rolling.observer-threshold=0
com.netflix.exhibitor.servers-spec=1\:10.10.0.230,2\:10.10.0.231,3\:10.10.0.232
com.netflix.exhibitor.cleanup-period-ms=300000
com.netflix.exhibitor.zookeeper-config-directory=/var/lib/dcos/exhibitor/conf
com.netflix.exhibitor.auto-manage-instances-fixed-ensemble-size=3
com.netflix.exhibitor.zookeeper-install-directory=/opt/mesosphere/active/exhibitor/usr/zookeeper
com.netflix.exhibitor.check-ms=30000
com.netflix.exhibitor.zookeeper-log-directory=/var/lib/dcos/exhibitor/zookeeper/transactions
com.netflix.exhibitor-rolling.auto-manage-instances=1
com.netflix.exhibitor-rolling.cleanup-period-ms=300000
com.netflix.exhibitor-rolling.zookeeper-config-directory=/var/lib/dcos/exhibitor/conf
com.netflix.exhibitor-rolling.auto-manage-instances-settling-period-ms=0
com.netflix.exhibitor-rolling.check-ms=30000
com.netflix.exhibitor.log-index-directory=/var/lib/dcos/exhibitor/zookeeper/transactions
com.netflix.exhibitor-rolling.log-index-directory=/var/lib/dcos/exhibitor/zookeeper/transactions
com.netflix.exhibitor.backup-period-ms=600000
com.netflix.exhibitor-rolling.connect-port=2888
com.netflix.exhibitor-rolling.election-port=3888
com.netflix.exhibitor-rolling.backup-extra=
com.netflix.exhibitor.client-port=2181
com.netflix.exhibitor-rolling.zoo-cfg-extra=tickTime\=2000&initLimit\=10&syncLimit\=5&quorumListenOnAllIPs\=true&maxClientCnxns\=0&autopurge.snapRetainCount\=5&autopurge.purgeInterval\=6
com.netflix.exhibitor-rolling.zookeeper-install-directory=/opt/mesosphere/active/exhibitor/usr/zookeeper
com.netflix.exhibitor-rolling.zookeeper-pid-path=
com.netflix.exhibitor.cleanup-max-files=20
com.netflix.exhibitor-rolling.auto-manage-instances-fixed-ensemble-size=3
com.netflix.exhibitor-rolling.backup-period-ms=600000
com.netflix.exhibitor.backup-max-store-ms=21600000
com.netflix.exhibitor-rolling.client-port=2181
com.netflix.exhibitor-rolling.cleanup-max-files=20
com.netflix.exhibitor-rolling.backup-max-store-ms=21600000
com.netflix.exhibitor.connect-port=2888
com.netflix.exhibitor.backup-extra=
com.netflix.exhibitor.observer-threshold=0
com.netflix.exhibitor.log4j-properties=
com.netflix.exhibitor.auto-manage-instances-apply-all-at-once=1
com.netflix.exhibitor.election-port=3888
com.netflix.exhibitor-rolling.auto-manage-instances-apply-all-at-once=1
com.netflix.exhibitor.zoo-cfg-extra=tickTime\=2000&initLimit\=10&syncLimit\=5&quorumListenOnAllIPs\=true&maxClientCnxns\=0&autopurge.snapRetainCount\=5&autopurge.purgeInterval\=6
com.netflix.exhibitor-rolling.zookeeper-log-directory=/var/lib/dcos/exhibitor/zookeeper/transactions
com.netflix.exhibitor.auto-manage-instances-settling-period-ms=0
com.netflix.exhibitor-rolling.log4j-properties=
com.netflix.exhibitor.auto-manage-instances=1
```

#### New UI Screen-shot Post update
image

File: exhibitor-core/src/main/resources/com/netflix/exhibitor/core/ui/index.html (hunk 1 above)
Comment (lingmann) [inline exhibitor-core/src/main/resources/com/netflix/exhibitor/core/ui/index.html]: Ok then, let's remove the 'Leave blank to use the ZooKeeper Snapshot Dir' description.

**Comment by BenWhitehead** [comment]
@jlingmann Message cleaned up, ready to merge. I can update dcos/dcos when the merge completes (already have a branch there for testing).

**Pull request merged**